service/controller/resource/snapshotid/create.go

```diff
@@ -0,0 +1,58 @@
+package snapshotid
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := meta.Accessor(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	i := &ec2.DescribeSnapshotsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String(fmt.Sprintf("tag:%s", key.TagCluster)),
+				Values: []*string{
+					aws.String(key.ClusterID(cr)),
+				},
+			},
+			{
+				Name: aws.String(fmt.Sprintf("tag:%s", key.TagSnapshot)),
+				Values: []*string{
+					aws.String(key.HAMasterSnapshotIDValue),
+				},
+			},
+		},
+	}
+
+	o, err := cc.Client.TenantCluster.AWS.EC2.DescribeSnapshots(i)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	count := len(o.Snapshots)
+	if count == 0 {
+		return microerror.Maskf(notExistsError, "snapshot")
+	}
+	if count != 1 {
+		return microerror.Maskf(executionFailedError, "expected one snapshot, got %d", count)
+	}
+
+	cc.Status.TenantCluster.MasterInstance.EtcdVolumeSnapshotID = key.HAMasterSnapshotIDValue
```

Comment:
if we find the ha master snapshot, store it in the controller context, other wise errors. @xh3b4sd is this the right place to be storing it in the controller context ?

Reply:
`key.HAMasterSnapshotIDValue` is not the snapshot ID. We need the real ID so we can create a real volume from it.

Reply:
yup, that was not the right thing to store. should be better now.